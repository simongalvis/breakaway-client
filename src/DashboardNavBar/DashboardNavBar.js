import React from 'react';
import { NavLink, BrowserRouter } from 'react-router-dom';
import './DashboardNavBar.css';
import logo from '../images/breakaway-logo.png'
import ApiContext from '../ApiContext';



class DashboardNavBar extends React.Component{
static contextType = ApiContext;


    render(){
        return(
            <header>
                <BrowserRouter>
                <NavLink to={Object.keys(this.context.loggedUser).length > 0 ? '/activities' : '/'}>
                    <img src={logo} alt='Blue breakaway logo' id='main-logo'/>
                </NavLink>
                <NavLink to="/new-activity">
                    <button className='nav-button' id='newactivity-breakaway-button' value='new activity'><b> +Breakaway!</b>  </button>
                </NavLink>
                </BrowserRouter>
                
                
            </header>
            
        )
    }
}


export default DashboardNavBar;