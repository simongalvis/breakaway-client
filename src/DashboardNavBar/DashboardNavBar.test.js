import React from 'react'
import ReactDOM from 'react-dom'
import DashboardNavBar from './DashboardNavBar'
import renderer from 'react-test-renderer';

describe(`DashboardNavbBar component`, () =>{
  it(`renders without crashing`, () =>{
    const div = document.createElement('div');
  
    ReactDOM.render(<DashboardNavBar/>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  })
  it(`Renders UI as expected`, () =>{
    const tree = renderer
      .create(<DashboardNavBar/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
