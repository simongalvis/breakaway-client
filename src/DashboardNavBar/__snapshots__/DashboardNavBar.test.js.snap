// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardNavbBar component Renders UI as expected 1`] = `
<header>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      alt="Blue breakaway logo"
      id="main-logo"
      src="breakaway-logo.png"
    />
  </a>
  <a
    aria-current={null}
    href="/new-activity"
    onClick={[Function]}
  >
    <button
      className="nav-button"
      id="newactivity-breakaway-button"
      value="new activity"
    >
      <b>
         +Breakaway!
      </b>
        
    </button>
  </a>
</header>
`;
